graphx/src/main/scala/org/apache/spark/graphx/lib/PageRank.scala

```diff
@@ -199,10 +199,10 @@ object PageRank extends Logging {
     require(sources.max <= Int.MaxValue.toLong,
       s"This implementation currently only works for source vertex ids at most ${Int.MaxValue}")
     val zero = Vectors.sparse(sources.size, List()).asBreeze
-    val sourcesInitMap = sources.zipWithIndex.map { case (vid, i) =>
+    val sourcesInitMap: Map[VertexId, BV[Double]] = sources.zipWithIndex.map { case (vid, i) =>
       val v = Vectors.sparse(sources.size, Array(i), Array(1.0)).asBreeze
       (vid, v)
-    }.toMap
+    }(collection.breakOut)
```

Comment:
Optimization nerds have already used it in the spark code:
https://github.com/apache/spark/blob/c4008480b781379ac0451b9220300d83c054c60d/sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala#L517